ouroboros-consensus/docs/ComponentDiagram.md

```diff
@@ -0,0 +1,70 @@
+This is a very high-level diagram  that indicates how the core data flow within the conceptual components of a single node's Consensus Layer.
+
+```mermaid
+graph LR
+
+    subgraph Consensus[Consensus Layer Components]
+        direction LR
+        Upstream["Upstream Peers Adapter\n& Mint"]---> |B| ChainSel --> |"U(H(B))"| Downstream[["Downstream Peer Adapters\n& Wallet"]]
+        ChainSel ---> |B| Diskpool -.-> |B| Downstream
+        Downstream --> |T| Mempool ---> |T| Upstream
+
+        ChainSel ---> |"H(L)"| Upstream
+        ChainSel --> |L| Mempool
+        ChainSel ---> |"O(L)"| Diskpool
+
+        Diskpool -.-> |B| ChainSel
+        Diskpool -.-> |"I(L)"| ChainSel
+    end
+
+    subgraph Legend
+        direction LR
+        Publisher --> |latest payload| Subscriber
+        Server -.-> |requested payload| Requester
+        X1[" "]
+        Y1[" "]
+        X1 --- |"T = transaction"| Y1
+        X1 --- |"B = block"| Y1
+        X1 --- |"L = ledger state"| Y1
+        X2[" "]
+        Y2[" "]
+        X2 --- |"H(X) = header part of X"| Y2
+        X2 --- |"U(X) = an update of X"| Y2
+        X3[" "]
+        Y3[" "]
+        X3 --- |"I(X) = only during intialization"| Y3
+        X3 --- |"O(X) = only occasionally"| Y3
+    end
+
+style Legend    fill:transparent,stroke:transparent
+style Consensus fill:transparent,stroke:transparent
+```
+
+The current implementation architecture is not an exact concretization of the conceptual components in this diagram.
+
+- The Diskpool component in this diagram is to blocks on disk as the Mempool component is to transactions in memory.
```

Comment:
Does the `Diskpool` only contain blocks? If so we should probably call this `BlocksPoolOnDisk` or something like that.